[TAS-59] Agregar carrusel a sección de proyectos

LUIS021201 opened the pull request:
## Describe your changes

Se agregó un carrusel para la sección de proyectos y se agregó un botón para el mismo.

## Task URL (Notion)


[[TAS-59] ](https://www.notion.so/Carrusel-en-secci-n-de-proyectos-87e01e67c10b47e2a6ea4281a4003e5e?pvs=4)

## What was done in this pull request

- [x] Agregar carrusel a la sección de proyectos
- [x] Agregar botón para carrusel

## Demo / Screenshot / Video

![image](https://github.com/CSIPro/csipro/assets/91043170/3d4a5314-778e-43fb-9390-d910cc7d5a0e)

LUIS021201 committed:
fix typo in projects.ts

LUIS021201 committed:
feat: Add carousel for projects section

Refactor the code to replace the individual project cards with a new ProjectsSection component. This component fetches projects from an API endpoint and displays them in a carousel. Additionally, lazy loading for images has been implemented to improve page load performance.

LUIS021201 committed:
feat: Add previous and next buttons to carousel

saulprl commented on src/app/page.tsx:
```diff
@@ -195,7 +168,8 @@ export default async function Home({
         <SectionTitle>Nuestros proyectos</SectionTitle>
-        <div className="flex w-full flex-col items-center gap-3 px-2 sm:flex-row sm:justify-center">
+        <ProjectsSection />
+        {/* <div className="flex w-full flex-col items-center gap-3 px-2 sm:flex-row sm:justify-center">
```
Delete this comment block

saulprl commented on src/components/projects-section/projects-section.tsx:
```diff
@@ -0,0 +1,66 @@
+  const projectsData = await projectsRes.json();
+
+  const projects = ProjectsResponse.safeParse(projectsData);
+  console.log(projects.error);
```
Se podría mejorar este console.log y return con una condicional. Si hay errores, hacer el `console.log` y generar una respuesta vacía. Si no, regresar los datos como normalmente.

saulprl reviewed (changes requested):
Cambios pequeños y listo

karolinaconk reviewed (dismissed):
+1000 papu :v

LUIS021201 committed:
feat: Remove commented out code in projects section

The commented out code in the projects section has been removed to improve code readability and maintainability.

saulprl reviewed (approved):
LGTM

LUIS021201 commented:
> +1000 papu :v

Gracias mamu UUUUUUUUhhh :V
